src/csharp/build_unitypackage.bat

```diff
@@ -72,6 +72,18 @@ copy /Y Grpc.Core\bin\Release\net45\System.Memory.dll unitypackage\unitypackage_
 @rem TODO(jtattermusch): also include XMLdoc
 copy /Y Grpc.HealthCheck\bin\Release\net45\Google.Protobuf.dll unitypackage\unitypackage_skeleton\Plugins\Google.Protobuf\lib\net45\Google.Protobuf.dll || goto :error
 
+@rem System.Memory 4.0.1.0 -> 4.0.1.1 #21908, #22251
```

Comment:
Interesting find, but this is a pretty dirty and fragile way of fixing.
Can we use a binding redirect in some way?

Also, we'd only need to ensure that the Google.Protobuf  dll is built with the same SDK as Grpc.Core.dll
(which seems doable). But I'd first like to understand how come Unity cannot handle references to slightly different versions of System.Memory? (this is quite normal on desktop .NET and .NET core, and the Grpc.Core and Google.Protobuf nugets interoperate without a problem).

Reply:
Unity3D has a feature that generates cpp code using a tool called il2cpp. At this time, the conflict occurs for the other two versions of the DLL. So I think Unity3D prevents other versions of the same DLL from being included.

Personally, I would like to modify Protobuf and gRPC to use the same SDK.

gRPC -> 4.5.2 or Protobuf -> 4.5.3

Reply:
ZCube, thanks for your work on driving to get this addressed!